packages/server/fileStorage/S3FileStoreManager.ts

```diff
@@ -98,17 +106,21 @@ export default class S3Manager extends FileStoreManager {
   async checkExists(key: string, assetDir?: FileAssetDir) {
     const Key = this.prependPath(key, assetDir)
     try {
-      await this.s3.send(new HeadObjectCommand({Bucket: this.baseUrl, Key}))
+      await this.s3.send(new HeadObjectCommand({Bucket: this.bucket, Key}))
     } catch (e) {
       if (e instanceof Error && e.name === 'NotFound') return false
+      else {
+        Logger.log(`Invalid error ${(e as Error).name}`)
+        throw new Error(`Can't verify if the object ${Key} is in the S3 bucket  ${this.bucket}`)
```

Comment:
-1 this will be a breaking change, it may work, but IIRC there was some case where s3 would error, but it would still exist. Since we're getting off S3 anyways, I say we keep the result the same (return true if error) but let's leave the log in so we can see under what case that's true